rotkehlchen/db/schema.py

```diff
@@ -437,6 +437,16 @@
 );
 """
 
+DB_CREATE_OPTIMISM_TRANSACTIONS = """
+CREATE TABLE IF NOT EXISTS optimism_transactions (
+    tx_hash BLOB NOT NULL,
+    chain_id INTEGER NOT NULL,
```

Comment:
You do not need the chain id. It's optimism (value is 10)

```suggestion
```

Reply:
pysqlcypher3/sqlite3 requires that all columns of the referenced table's primary key be in the foreign key, right? I can't eliminate the chain_id without getting a foreign key error:

      rotkehlchen/tests/unit/test_evm_tx_decoding.py:98: in _add_transactions_to_db
          dboptimismtx.add_evm_transactions(cursor, [transaction_opt], relevant_address=ethereum_accounts[0])  # noqa: E501
      rotkehlchen/db/optimismtx.py:45: in add_evm_transactions
          self.db.write_tuples(
      rotkehlchen/db/dbhandler.py:1537: in write_tuples
          write_cursor.executemany(query, tuples)
      _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
      
      self = <rotkehlchen.db.drivers.gevent.DBCursor object at 0x7fc9e631f520>
      statement = '\n            INSERT INTO optimism_transactions(\n              tx_hash,\n              l1_fee)\n            VALUES (?, ?)\n        '
      bindings = ([(HexBytes('0x063d45910f29e0954a52aee39febba9be784d49af7588a590dc2fd7d156b4665'), 455063072063200)],)
      
          def executemany(self, statement: str, *bindings: Sequence[Sequence]) -> 'DBCursor':
              if __debug__:
                  logger.trace(f'EXECUTEMANY {statement}')
      >       self._cursor.executemany(statement, *bindings)
      E       pysqlcipher3.dbapi2.OperationalError: foreign key mismatch - "optimism_transactions" referencing "evm_transactions"


This is the last thing I need to figure out before I'm ready to do some final linting and push changes

Reply:
> pysqlcypher3/sqlite3 requires that all columns of the referenced table's primary key be in the foreign key, right? I can't eliminate the chain_id without getting a foreign key error:
> 
> ```
>   rotkehlchen/tests/unit/test_evm_tx_decoding.py:98: in _add_transactions_to_db
>       dboptimismtx.add_evm_transactions(cursor, [transaction_opt], relevant_address=ethereum_accounts[0])  # noqa: E501
>   rotkehlchen/db/optimismtx.py:45: in add_evm_transactions
>       self.db.write_tuples(
>   rotkehlchen/db/dbhandler.py:1537: in write_tuples
>       write_cursor.executemany(query, tuples)
>   _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
>   
>   self = <rotkehlchen.db.drivers.gevent.DBCursor object at 0x7fc9e631f520>
>   statement = '\n            INSERT INTO optimism_transactions(\n              tx_hash,\n              l1_fee)\n            VALUES (?, ?)\n        '
>   bindings = ([(HexBytes('0x063d45910f29e0954a52aee39febba9be784d49af7588a590dc2fd7d156b4665'), 455063072063200)],)
>   
>       def executemany(self, statement: str, *bindings: Sequence[Sequence]) -> 'DBCursor':
>           if __debug__:
>               logger.trace(f'EXECUTEMANY {statement}')
>   >       self._cursor.executemany(statement, *bindings)
>   E       pysqlcipher3.dbapi2.OperationalError: foreign key mismatch - "optimism_transactions" referencing "evm_transactions"
> ```
> 
> This is the last thing I need to figure out before I'm ready to do some final linting and push changes

Hmm this is an ugly dependecy. But then make it a computed column so that it does not take space. Much like we do with the nfts table and the blockchain column.

```
    blockchain TEXT GENERATED ALWAYS AS ('ETH') VIRTUAL,
```

Note to use single quote in the sqlite query itself.

https://www.sqlite.org/gencol.html